foster55f/market2table-fe
# Vendor link and Vendor Edit update and implementation

### What does this PR do?
- This PR implements POSTs for the VendorForm to POST a new vendor and then post vendor products to that newly created vendor
- Creates VendorLinkContainer, VendorMarketList, VendorMarketSearch, MarketListCard components
- Creates container for linking vendors to markets.
- Creates section with a form to input zip code and GET all markets from that zip code - displays those markets with a button to add the market to the vendor market List
- Creates section that gets all markets based off a vendorId and displays those markets with the ability to remove it DELETE.
- When a market is added it is removed from the markets available to add list and added to the markets linked list.
- error handling for the zip code form.

### How to test?
 - Pull down branch
 - run 'npm start' and the site should open in browser via localhost:3000
 - run 'npm install'
- click Link To Markets button and search via zip and add/remove markets

### Issues:
The Create Vendor Form when in the edit pre-existing vendor is not currently functional and will probably create a duplicate instead of updating the pre-existing one.

### Notes:
- Might still need some styling and needs to be responsive.

### Relevant Screenshots:

<img width="1440" alt="Screen Shot 2020-02-24 at 3 45 41 PM" src="https://user-images.githubusercontent.com/49846853/75198227-5631e500-571d-11ea-98f1-ee0ad71eb623.png">

<img width="1440" alt="Screen Shot 2020-02-24 at 3 45 49 PM" src="https://user-images.githubusercontent.com/49846853/75198237-5b8f2f80-571d-11ea-85a4-7632d5b321f9.png">
